### DISCUSSION TITLE
LoTW Certificate Upload Problems

### DISCUSSION THREAD

**Discussion by rjmusto** [magicbug/Cloudlog#1096] [General]
Hi,
New user here, using the hosted version of Cloudlog.

After uploading my backlog of QSO's, I also uploaded the LoTW p12 certificate and did a sync. All ok.

Now though it is telling me again that I need to upload a p12 certificate - but every time I try I'm bouced back to the start asking to upload a certificate. So can't sync to LoTW anymore.

So a bit stuck on what to do here.

Thanks,
Ralph

**Comment by magicbug** [comment]
Can you provide some screenshots Ralph be interesting to see whats happening? also is the certificate password protected as Cloudlog wont handle those.

Peter.

**Reply by rjmusto** [reply]
Hi Peter,
If I select LoTW Sync from the callsign menu then I get taken to the first image, I then click on Upload Cert and get to the second image. But when clicking Uplopad File, I end up back at the first image, with nothing else reported.
![Screenshot 2021-07-22 at 20 19 58](https://user-images.githubusercontent.com/35890758/126697545-78ae5d40-4af3-4d45-b964-40ca634841f6.png)
![Screenshot 2021-07-22 at 20 20 24](https://user-images.githubusercontent.com/35890758/126697576-a653aaf1-ac47-4e0b-96fc-53747bd2a776.png)

**Reply by magicbug** [reply]
OK thats helpful, could be a couple of things

- .p12 is password protected
- system file permissions aren't setup you can check that by going to Admin -> Debug (should all be green)
- Theres some other bug in the code I don't know about!

**Reply by rjmusto** [reply]
System permissions are all green.

Rather annoyingly, the p12 certificate was generated when i had a previous version of OSX installed, so can’t run tqsl to check it without requesting a new certificate.
(I’ll dig out a backup disk I have though and try booting from that, then i might be able to run tqsl ok).

The odd part of this though is that it all worked fine the first time i did it - it synced with LoTW just fine. Now it won’t.

R

**Reply by call-me-amy** [reply]
Did you find a solution?  I'm having the same problem.

**Reply by call-me-amy** [reply]
openssl_pkcs12_read is giving error "error:0308010C:digital envelope routines::unsupported"

I'm assuming a newer version of openssl doesn't like something about the LOTW certificate.

**Comment by rayslinky** [comment]
I had the [same issue with tqsl](https://groups.arrl.org/g/ARRL-LoTW/topic/tqsl_2_6_1_linux_version/91067801?p=,,,20,0,0,0::recentpostdate/sticky,,,20,0,0,91067801,previd%3D1653646864998983328,nextid%3D1651278596071380803&previd=1653646864998983328&nextid=1651278596071380803) a few months ago, and then they patched it.

**Comment by call-me-amy** [comment]
Needed the newer version of tsql, and needed to enable legacy_sect support in openssl on the Cloudlog instance.
